### PR TITLE
Unity android build obsolete warning

### PR DESCRIPTION
**Pull request by olmisa**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Providing Android resources from the Unity 'Plugins' folder was deprecated. The code was reworked. Now the application settings file creates as 'AppCenterLoader' library resource.

## Related PRs or issues

[AB#72139](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/72139)
https://github.com/microsoft/appcenter-sdk-unity/issues/356

### CODE REVIEW

File: Assets/AppCenter/Editor/AppCenterPreBuild.cs [hunk 1]
```diff
@@ -49,6 +53,48 @@ public void OnPreprocessBuild(BuildTarget target, string path)
...
+        AndroidLibraryHelper.UnzipFile(loaderZipFile, loaderFolder);
+        if (!Directory.Exists(loaderFolder))
+        {
+            Debug.LogWarning("Unzipping loader folder failed.");
+            return;
+        }
+
+        settingsMaker.CommitSettings(settingsFilePath);
+
+        // Delete the AndroidManifest.xml.meta file if generated
```
Comment (annakocheshkova) [inline Assets/AppCenter/Editor/AppCenterPreBuild.cs]: Fix the comment

Comment (olmisa) [reply Assets/AppCenter/Editor/AppCenterPreBuild.cs]: Fixed

File: Assets/AppCenter/Editor/AndroidLibraryHelper.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
+﻿using System.Diagnostics;
```
Comment (guperrot) [inline Assets/AppCenter/Editor/AndroidLibraryHelper.cs]: copyright missing

File: Assets/AppCenter/Editor/AppCenterPreBuild.cs [hunk 3]
```diff
@@ -49,6 +55,48 @@ public void OnPreprocessBuild(BuildTarget target, string path)
         }
     }
 
+#if UNITY_ANDROID
+    public static void AddSettingsFileToLoader(AppCenterSettingsMakerAndroid settingsMaker)
+    {
+        var loaderZipFile = AppCenterSettingsContext.AppCenterPath + "/Plugins/Android/appcenter-loader-release.aar";
+        var loaderFolder = "appcenter-loader-release";
+        var settingsFilePath = "appcenter-loader-release/res/values/appcenter-settings.xml";
```
Comment (guperrot) [inline Assets/AppCenter/Editor/AppCenterPreBuild.cs]: folder name could be appended from loaderFolder, and maybe use constants for all of these paths

File: Assets/AppCenter/Editor/AppCenterPreBuild.cs [hunk 4]
```diff
@@ -49,6 +55,48 @@ public void OnPreprocessBuild(BuildTarget target, string path)
...
+        var settingsMetaFilePath = "appcenter-loader-release/res/values/appcenter-settings.xml.meta";
+
+        if (!File.Exists(loaderZipFile))
+        {
+            Debug.LogWarning("Failed to load dependency file appcenter-loader-release.aar");
+            return;
+        }
+
+        // Delete unzipeed directory if it already exists.
```
Comment (guperrot) [inline Assets/AppCenter/Editor/AppCenterPreBuild.cs]: typo
```suggestion
        // Delete unzipped directory if it already exists.
```

File: Assets/AppCenter/Editor/AppCenterPreBuild.cs [hunk 5]
```diff
@@ -49,6 +55,48 @@ public void OnPreprocessBuild(BuildTarget target, string path)
...
+        AndroidLibraryHelper.UnzipFile(loaderZipFile, loaderFolder);
+        if (!Directory.Exists(loaderFolder))
+        {
+            Debug.LogWarning("Unzipping loader folder failed.");
+            return;
+        }
+
+        settingsMaker.CommitSettings(settingsFilePath);
+
+        // Delete the appcenter-settings.xml.meta file if generated
```
Comment (guperrot) [inline Assets/AppCenter/Editor/AppCenterPreBuild.cs]: End comments with dots.
```suggestion
        // Delete the appcenter-settings.xml.meta file if generated.
```

File: Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs [hunk 6]
```diff
@@ -6,7 +6,7 @@
 
 public class AppCenterSettingsMakerAndroid : IAppCenterSettingsMaker
 {
-    private const string AppCenterResourcesFolderPath = "Assets/Plugins/Android/res/values/";
+    private const string AppCenterResourcesFolderPath = "AppCenterLoaderApp/appcenter-loader/src/main/res/values/";
```
Comment (guperrot) [inline Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs]: the path constants should be shared with the other file

**Review by guperrot — CHANGES_REQUESTED** [review]
What kind of tests were done for this as this changes integration build a bit. Did we verify a blank app and demo upgrade?

**Comment by olmisa** [comment]
The changes were verified on a demo and on the blank app.

File: Assets/AppCenter/Editor/AppCenterPreBuild.cs [hunk 7]
```diff
@@ -58,18 +58,18 @@ public void OnPreprocessBuild(BuildTarget target, string path)
 #if UNITY_ANDROID
     public static void AddSettingsFileToLoader(AppCenterSettingsMakerAndroid settingsMaker)
     {
-        var loaderZipFile = AppCenterSettingsContext.AppCenterPath + "/Plugins/Android/appcenter-loader-release.aar";
-        var loaderFolder = "appcenter-loader-release";
-        var settingsFilePath = "appcenter-loader-release/res/values/appcenter-settings.xml";
-        var settingsMetaFilePath = "appcenter-loader-release/res/values/appcenter-settings.xml.meta";
+        string loaderZipFile = AppCenterSettingsContext.AppCenterPath + "/Plugins/Android/appcenter-loader-release.aar";
```
Comment (guperrot) [inline Assets/AppCenter/Editor/AppCenterPreBuild.cs]:
```suggestion
        var loaderZipFile = AppCenterSettingsContext.AppCenterPath + "/Plugins/Android/appcenter-loader-release.aar";
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,6 +2,12 @@
 
 ## Version 2.6.1 (Under development)
 
+### App Center
+
+#### Android
+
+* **[Fix]** Fix the Unity warning that appears during the building app for Android.
```
Comment (guperrot) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Fix a Unity warning that appears when building for Android.
```

**Review by guperrot — APPROVED** [review]

File: Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs [hunk 9]
```diff
@@ -6,8 +6,8 @@
 
 public class AppCenterSettingsMakerAndroid : IAppCenterSettingsMaker
 {
-    private const string AppCenterResourcesFolderPath = "Assets/Plugins/Android/res/values/";
-    private const string AppCenterResourcesPath = AppCenterResourcesFolderPath + "appcenter-settings.xml";
+    public const string AppCenterLoaderResourcesFolderPath = "AppCenterLoaderApp/appcenter-loader/src/main/res/values/";
+    public const string AppCenterLoaderResourcesPath = AppCenterLoaderResourcesFolderPath + "appcenter-settings.xml";
```
Comment (MatkovIvan) [inline Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs]: This is totally incorrect. @olmisa please revert it

File: Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs [hunk 10]
```diff
@@ -6,8 +6,8 @@
 
 public class AppCenterSettingsMakerAndroid : IAppCenterSettingsMaker
 {
-    private const string AppCenterResourcesFolderPath = "Assets/Plugins/Android/res/values/";
-    private const string AppCenterResourcesPath = AppCenterResourcesFolderPath + "appcenter-settings.xml";
+    public const string AppCenterLoaderResourcesFolderPath = "AppCenterLoaderApp/appcenter-loader/src/main/res/values/";
```
Comment (MatkovIvan) [inline Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs:9]: It doesn't work. It must not write to this location, it doesn't make any sense if there are no SDK sources in project's location

**Pull request merged**